lib/clientv2/networkpolicy.go

```diff
@@ -51,23 +50,26 @@ func (r networkPolicies) Create(ctx context.Context, res *apiv2.NetworkPolicy, o
 // Update takes the representation of a NetworkPolicy and updates it. Returns the stored
 // representation of the NetworkPolicy, and an error, if there is any.
 func (r networkPolicies) Update(ctx context.Context, res *apiv2.NetworkPolicy, opts options.SetOptions) (*apiv2.NetworkPolicy, error) {
-	out, err := r.client.resources.Update(ctx, opts, apiv2.KindNetworkPolicy, r.namespace, res)
+	out, err := r.client.resources.Update(ctx, opts, apiv2.KindNetworkPolicy, res)
 	if out != nil {
 		return out.(*apiv2.NetworkPolicy), err
 	}
 	return nil, err
 }
 
 // Delete takes name of the NetworkPolicy and deletes it. Returns an error if one occurs.
-func (r networkPolicies) Delete(ctx context.Context, name string, opts options.DeleteOptions) error {
-	err := r.client.resources.Delete(ctx, opts, apiv2.KindNetworkPolicy, r.namespace, name)
-	return err
+func (r networkPolicies) Delete(ctx context.Context, namespace, name string, opts options.DeleteOptions) (*apiv2.NetworkPolicy, error) {
```

Comment:
thinking out loud, what if we just have the clients make sure namespace is set in the resource before they make a Get or Delete call, so we can just extract the namespace from ` r.namespace`? This way we don't have to change the function signature for namespaced resources

Reply:
there isn't a resource on a Get or Delete call.

Reply:
Doh, yes